compiler/rustc_hir_typeck/src/op.rs

```diff
@@ -521,6 +521,63 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         }
                     }
                 }
+
+                // Suggest using `add`, `offset` or `offset_from` for pointer - {integer},
+                // pointer + {integer} or pointer - pointer.
+                if op.span.can_be_used_for_suggestions() {
+                    if op.node == hir::BinOpKind::Add {
+                        match (lhs_ty.kind(), rhs_ty) {
+                            (RawPtr(..), rhs_ty) if rhs_ty.is_integral() => {
```

Comment:
this can be an `if lhs_ty.is_unsafe_ptr() && rhs_ty.is_integral() {` (and similarly below)